src/util/range.h

```diff
@@ -0,0 +1,348 @@
+/*******************************************************************\
+
+Module: Range
+
+Author: Romain Brenguier, romain.brenguier@diffblue.com
+
+\*******************************************************************/
+
+/// \file
+/// Ranges: pair of begin and end iterators.
```

Comment:
I think this could do with a bit more text to explain _why_ would want to use this - there is an elaborate and very useful explanation below, but this sentence by itself is not particularly helpful.